include/scippp/model.hpp

```diff
@@ -250,6 +251,26 @@ class Model {
         }
     }
 
+    /**
+     * Writes original problem to file.
+     *
+     * @since 1.1.0
+     * @param filename output file name including extension
+     * @param genericNames using generic variable and constraint names?
+     * @attention Do not use an std::string or std::filesystem::path as argument \p filename,
+     *            as this will call the other overload instead!
+     */
+    void writeOrigProblem(const std::filesystem::directory_entry& filename, bool genericNames = false) const;
+
+    /**
+     * Writes original problem to standard output.
+     *
+     * @since 1.1.0
+     * @param extension file extension to derive the output format from
+     * @param genericNames using generic variable and constraint names?
+     */
+    void writeOrigProblem(const std::string& extension, bool genericNames = false) const;
```

Comment:
Would it make sense to define an enum for the available file extensions? The usual SCIP interface doesn't seem to offer any such thing.

This would also make the warning for the overloaded function obsolete.

Reply:
SCIP solves it as follows: depending on the extension, it loops over all registered plugins and asks them whether they support that, if so, it gives the task to the right plugin. That means: there are extensions only suitable for certain problem types, and the user can add other plugins that support other extensions. Creating such an enum would reflect only the default plugins of SCIP

Reply:
how to deal with that? I could provide the current list of extensions for the default plugins. But then the user has no SCIP++ way to use own plugins. On the other hand, the user has to use the C way to even add own plugins at the moment ...

Reply:
In that case: Fine to keep it.